crates/proof-of-sql/src/base/database/owned_table.rs

```diff
@@ -130,6 +130,11 @@ impl<S: Scalar> OwnedTable<S> {
     pub fn column_names(&self) -> impl Iterator<Item = &Ident> {
         self.table.keys()
     }
+    /// Returns the column with the given position.
+    #[must_use]
+    pub fn column(&self, index: usize) -> Option<OwnedColumn<S>> {
+        self.table.values().nth(index).cloned()
+    }
```

Comment:
Would it be fine for your uses to have this return a reference? I also think it should probably be more clear in the name that it's looking up by index, rather than by column name.
```suggestion
    pub fn column_index(&self, index: usize) -> Option<&OwnedColumn<S>> {
        self.table.get_index(index).map(|(_, v)| v)
    }
```